corehq/apps/groups/views.py

```diff
@@ -28,7 +29,16 @@ def add_group(request, domain):
             "We could not create the group; "
             "please give it a name first"
         ))
-        return HttpResponseRedirect(request.META['HTTP_REFERER'])
+        redirection_url = request.META['HTTP_REFERER']
```

Comment:
I think the thing we want to check for here isn't Server Side Request Forgery—with the code as written, the server already won't make a request to this URL, the client's browser will. I think what we want to protect against is redirecting to another website, such as a phishing site that looks a lot like commcare. For that, we need to check that we never redirect to a site other than our own site's domain, which you can get with `get_url_base`, (or start with `/` and are thus relative to the current domain, but I don't think that would ever show up in the `Referer` header).

Reply:
Hey, do you mean to apply this check on client side?  One point. Since the server is the one triggering the redirect yeah, doesn't it make more sense to handle that on the server side and not on the client-side? 
Putting the security check on the client side may not be the best option here as this is not redirection caused by user's action(click on some URL on the page). It is rather a redirection triggered by the Backend.

Reply:
I didn't mean to check for it on the client side. This is the correct place for the check, but I'm saying I think this code is checking for the wrong thing.

The check you put in place here would make it so that if I send in the header `Referer: localhost` it will fail, but if I send in the header `Referer: example.com` or `Referer: mymalicioussite.com` or `Referer: commcarehq.org.mymalicioussite.com` (from which I could run a phishing operation that looks like commare) this attacker would succeed. So I'm saying you need a different check here that makes sure it starts with www.commcarehq.org (or the appropriate site, taken from `get_url_base`) instead of `is_url_or_host_banned`.

Reply:
@dannyroberts  used the base url check now. Can you check if this looks good